modules/ossm-uguide-routing-ingress.adoc

```diff
@@ -0,0 +1,228 @@
+
+[id="ossm-routing-ingress_{context}"]
+= Managing ingress traffic
+
+In a service mesh, you can use a {ProductName} gateway to enable {ProductShortName} features such as monitoring and route rules to be applied to traffic entering the cluster.
+
+You can configure {ProductShortName} to expose a service outside of the service mesh using an {ProductShortName} gateway.
+
+.Before you begin
+
+* Install {ProductShortName}.
+
+* Change to the the `istio` directory.
+
+[id="ossm-routing-determine-ingress_{context}"]
+== Determining the ingress IP and ports
+
+Run the following command to determine if your Kubernetes cluster is running in an environment that supports external load balancers:
+
+----
+$ oc get svc istio-ingressgateway -n istio-system
+----
+
+That command returns the `NAME`, `TYPE`, `CLUSTER-IP`, `EXTERNAL-IP`,    `PORT(S)`, and `AGE` of each item in your namespace. 
+
+If the `EXTERNAL-IP` value is set, your environment has an external load balancer that you can use for the ingress gateway. 
+
+If the `EXTERNAL-IP` value is `<none>`, or perpetually `<pending>`, your environment does not provide an external load balancer for the ingress gateway. You can access the gateway using the service's https://kubernetes.io/docs/concepts/services-networking/service/#nodeport[node port].
+
+Choose the instructions for your environment:
+
+.External load balancer
+
+Follow these instructions if your environment has an external load balancer.
+
+Set the ingress IP and ports:
+
+----
+$ export INGRESS_HOST=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+----
+
+----
+$ export INGRESS_PORT=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+----
+
+----
+$ export SECURE_INGRESS_PORT=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+----
+
+In some environments, the load balancer may be exposed using a host name instead of an IP address. For that case, the ingress gateway's `EXTERNAL-IP` value is not be an IP address. Instead, it's a host name, and the previous command fails to set the `INGRESS_HOST` environment variable.
+
+Use the following command to correct the `INGRESS_HOST` value:
+
+----
+$ export INGRESS_HOST=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+----
+
+.Node port
+
+Follow these instructions if your environment does not have an external load balancer. You must use a node port instead.
+
+Set the ingress ports:
+
+----
+$ export INGRESS_PORT=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+----
+
+----
+$ export SECURE_INGRESS_PORT=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+----
+
+Setting the ingress IP depends on the cluster provider:
```

Comment:
Do we want to supply information for GKE (do we support it?), Minikube (we definitely don't support it), Docker for desktop (I'm pretty sure we don't support it)?